src/simulation.hpp

```diff
@@ -2054,14 +2065,49 @@ template <typename problem_t> void AMRSimulation<problem_t>::InitParticles()
 	}
 }
 
-template <typename problem_t> void AMRSimulation<problem_t>::InitCICParticles()
+template <typename problem_t> void AMRSimulation<problem_t>::InitPhyParticles()
 {
+	if (do_rad_particles != 0) {
+		AMREX_ASSERT(RadParticles == nullptr);
+
+		// Create radiating particle descriptor
+		// the indices are: mass, lum, hydro_interact
+		auto radParticleDesc = std::make_unique<quokka::PhysicsParticleDescriptor>(-1, quokka::RadParticleLumIdx, false);
+
+		// Create particle container
+		RadParticles = std::make_unique<quokka::RadParticleContainer<problem_t>>(this);
+		RadParticles->SetVerbose(0);
+
+		// Set container in descriptor
+		radParticleDesc->neighborParticleContainer_ = RadParticles.get();
```

Comment:
This does not transfer ownership of `RadParticles`. In your implementation, `RadParticles` will be deallocated on line 2091, and attempting to access `radParticleDesc->neighborParticleContainer_` will cause a segmentation fault.

This is why I am so insistent about using `std::unique_ptr` instead of raw pointers. You cannot have this bug if `radParticleDesc->neighborParticleContainer` is a `std::unique_ptr`.